File: backend/app/schemas/topic.py
from pydantic import BaseModel


class TopicBase(BaseModel):
    name: str
    description: str
    keywords: list[str]
    schedule: str
    enabled: bool = True


class TopicCreate(TopicBase):
    pass


class TopicUpdate(TopicBase):
    pass


class TopicRead(TopicBase):
    id: int

    model_config = {"from_attributes": True}

File: frontend/src/types/topic.ts
// Topic types
export interface Topic {
  id: number
  name: string
  description: string
  keywords: string[]
  schedule: string
  enabled: boolean
}

export interface TopicCreatePayload {
  name: string
  description: string
  keywords: string[]
  schedule: string
  enabled: boolean
}
